gems/pending/util/miq-password.rb

```diff
@@ -122,16 +122,24 @@ def self.clear_keys
     @v2_key = @v1_key = @v0_key = nil
   end
 
+  def self.all_keys
+    v2_key ? [v2_key] + legacy_keys : legacy_keys
+  end
+
   def self.v2_key
-    @v2_key ||= ez_load("#{key_root}/v2_key")
+    @v2_key ||= ez_load("v2_key")
+  end
+
+  def self.legacy_keys
+    [v1_key, v0_key].delete_if(&:!)
```

Comment:
Alternately:

```ruby
[v1_key, v0_key].select { |k| k }
```

Seems clearer to me anyway as it does't have a double negative (i.e. delete and !)